web/src/main/java/org/mskcc/cbio/oncokb/api/pub/v1/TrialsApiController.java

```diff
@@ -0,0 +1,360 @@
+            AWSCredentials credentials = new BasicAWSCredentials(s3AccessKey, s3SecretKey);
+            AmazonS3 s3client = AmazonS3ClientBuilder.standard()
+                .withCredentials(new AWSStaticCredentialsProvider(credentials))
+                .withRegion(Regions.US_EAST_1)
+                .build();
+            S3Object s3object = s3client.getObject("oncokb", "drug-matching/result.json");
+            S3ObjectInputStream inputStream = s3object.getObjectContent();
+            JSONParser jsonParser = new JSONParser();
+            JSONObject jsonObject = (JSONObject)jsonParser.parse(new InputStreamReader(inputStream, "UTF-8"));
+            
+            Set<String> cancerTypes = new HashSet<>(Arrays.asList(body.getCancerTypes()));
+            if (cancerTypes.contains(SpecialTumorTypeName.ALL_TUMORS.getValue())){
+                List<Trial> trials = new ArrayList<>();
+                Set<String> nctIDSet = new HashSet<>();
+                for (Object item: jsonObject.keySet()){
+                    String oncoTreeCode = (String)item;
+                    Tumor tumor = getTumor(jsonObject, oncoTreeCode);
+                    for (Trial curTrial: tumor.getTrials()){
+                        if (!nctIDSet.contains(curTrial.getNctId())){
+                            nctIDSet.add(curTrial.getNctId());
+                            trials.add(curTrial);
+                        }
+                    }
+                }
+                result.put(SpecialTumorTypeName.ALL_TUMORS.getValue(), trials.toArray(new Trial[trials.size()]));
+                return new ResponseEntity<>(result, status);
+            }
+
+            for (String cancerType: cancerTypes){
+                Set<String> nctIDSet = new HashSet<>();
+                List<Trial> addTrials = new ArrayList<>();
+                List<Trial> trials = new ArrayList<>();
+                if (isSpecialTumorType(cancerType)){
+                    trials = Arrays.asList(getTrialsForSpecialCancerTye(jsonObject, cancerType));
+                }
+                else{
+                    trials = getTrialsByCancerType(jsonObject, cancerType);
+                }
+                for (Trial trial: trials){
+                    if (!nctIDSet.contains(trial.getNctId())){
+                        nctIDSet.add(trial.getNctId());
+                        addTrials.add(trial);
+                    }
+                }
+                result.put(cancerType, addTrials.toArray(new Trial[addTrials.size()]));
+            }
+
+            if (result.isEmpty()){
+                status = HttpStatus.NOT_FOUND;
+                return new ResponseEntity<>(result, status);
+            }
+        }
+        return new ResponseEntity<>(result, status);
+    }
+
+    private Tumor getTumor(JSONObject jsonObject, String oncoTreeCode){
+        Tumor tumor = new Tumor();
+        if (jsonObject.containsKey(oncoTreeCode)){
+            JSONObject tumorObj = (JSONObject)jsonObject.get(oncoTreeCode);

```

Comment:
You can convert the json to model directly. https://stackoverflow.com/questions/24231223/how-can-i-cast-a-jsonobject-to-a-custom-java-class